Release Candidate r2.1

PedroDiez opened the pull request:
#### What type of PR is this?

* documentation
* subproject management


#### What this PR does / why we need it:

Release 2.1 (Release candidate) PR for Customer Insights within Fall25 MetaRelease (M3)


#### Which issue(s) this PR fixes:

Fixes #35
Fixes #36 
Fixes #37 
Fixes #38 

#### Special notes for reviewers:

Issue #35 is covered specifically in PR #43. Already covered and merged
Issue #36 is covered in this PR
Issue #37 has been already covered in PRs #29 and #40
Issue #38 is covered specifically in PR #44. Already covered and merged

Therefore with regards to API Spec and Gherkin Feature in this PR is reflected "cosmetic" changes as PR#43 and PR#44 are needed to be merged in advance and then align this PR with main


#### Changelog input

```
 Release Candidate r2.1 for Customer Insights API

```

#### Additional documentation 

N/A

hdamker commented:
@PedroDiez would you mind to put the PR into draft status until it is ready for review (ie the two PRs are merged)?

PedroDiez commented:
> Hi @PedroDiez , @hdamker - just to check my understanding on two items:
> 
> 1. the `IDENTIFIERS_MISMATCH` error you have is distinct from the device-related IDENTIFER_MISMATCH that we have now removed?
> 
> ```yaml
>             GENERIC_422_IDENTIFIERS_MISMATCH:
>               description: Error when provided `idDocument` is not consistent with the phone number.
>               value:
>                 status: 422
>                 code: CUSTOMER_INSIGHTS.IDENTIFIERS_MISMATCH
>                 message: Provided `idDocument` is not consistent with the phone number.
> ```
> 
> 2. The API takes an optional `phoneNumber` in the request, but not a `Device` object.  Is it correct that the `DeviceResponse` object is therefore not required?

Hi

1. Yes, this `CUSTOMER_INSIGHTS.IDENTIFIERS_MISMATCH` represents a specific one for Customer insights API not the transversal one defined in Commonalities (in Spring 25 and now removed since Fall25).

This exception is for instance for a 2-legged scenario, phoneNumber is indicated, when idDocument is also needed for API Provider logic. In such a case both parameters are not consistent (e.g. idDocument is not associated to the phoneNumber) that exception is triggered.

2. In my understanding it is OK and not needed, because this API has not the device object concept and that `DeviceResponse` is applicable for APIs that own such concept (References https://github.com/camaraproject/Commonalities/issues/431 // https://github.com/camaraproject/Commonalities/pull/455)

Kevsy reviewed (dismissed):
Release management review made (see https://github.com/camaraproject/ReleaseManagement/issues/252) and LGTM.

KevScarr reviewed (dismissed):
All approved.

hdamker commented:
Some minor topics, maybe worth to open issues for coming versions:

- Test file doesn't validate all error codes defined in API spec. Missing: `CUSTOMER_INSIGHTS.ID_DOCUMENT_NOT_SUPPORTED`
`CUSTOMER_INSIGHTS.ID_DOCUMENT_REQUIRED`
- there is mix of two different tagging patterns in the test definition: `@retrieve_scoring_C02.01` and `@retrieve_scoring_422.01`

hdamker commented on code/API_definitions/customer-insights.yaml:
```diff
@@ -78,17 +78,17 @@ info:
-  x-camara-commonalities: 0.5
+  x-camara-commonalities: 0.6
 externalDocs:
   description: Product documentation at Camara
```
An easy one:

```suggestion
  description: Product documentation at CAMARA
```

rartych commented:
Just for the note for future discussion: as this API is quite close to KYC, it can follow their convention for error codes:
```
                      - KNOW_YOUR_CUSTOMER.ID_DOCUMENT_REQUIRED
                      - KNOW_YOUR_CUSTOMER.ID_DOCUMENT_MISMATCH
```

PedroDiez commented:
Hi @hdamker, @rartych thanks for the comments.

I will open issues to look for a new exception wording for CUSTOMER_INSIGHTS.IDENTIFIERS_MISMATCH and fix the pending tests. I think it is worth to have it for the public release.

Regarding this point 

> there is mix of two different tagging patterns in the test definition: @retrieve_scoring_C02.01 and @retrieve_scoring_422.01

I think there is no issue under this and the mixing is a possible scenario.

Based on Commonalities artifacts https://github.com/camaraproject/Commonalities/tree/main/artifacts/testing, the wording for applicable tests of re-usable [Series C02](https://github.com/camaraproject/Commonalities/blob/main/artifacts/testing/C02-phoneNumber-errors.feature) is kept and within the Gherkin .feature error tests are grouped by "status". That is the reason to have the two tagging within a "same section" of the .feature

PedroDiez commented:
> > I will open issues to look for a new exception wording for CUSTOMER_INSIGHTS.IDENTIFIERS_MISMATCH and fix the pending tests. I think it is worth to have it for the public release.
> 
> HI @PedroDiez , just to confirm will you be submitting these issues/commits for Fall 25?

Hi @Kevsy, yes i will open this week to consider them for the public release of this API in Fall 25

hdamker commented:
@PedroDiez @Kevsy the release should be created immediately after the release PR is merged ... or otherwise round: if you don't have the time to create the release, don't merge the PR.

PedroDiez commented:
> @PedroDiez @Kevsy the release should be created immediately after the release PR is merged ... or otherwise round: if you don't have the time to create the release, don't merge the PR.

Hi,
Ok understood Herbert, thanks for the comment. 
We will proceed in that way.
